### DISCUSSION TITLE
Multi-line triggers

### DISCUSSION THREAD

**Discussion by rigrig** [scandum/tintin#165] [Beta]
This is pretty awesome!

Why the need to include `\n` for enabling them though, would it be possible to just make triggers containing `%a` multiline?
So this would just work: `#action {\A%* tells you: %a\Z} {/handle_tell {%1} {%2}}`

Another nice thing would be if there was some flag to make a trigger match newlines as space.
So `#act {line containing some text} {/some_text}` would trigger on
```
This is a (variable length) server-side wrapped line containing some
text.
```
without needing to resort to `#act {line%acontaining%asome%atext} {/some_text}`

**Comment by scandum** [comment]
This should work:
```
#action {\A%* tells you: %a\n\Z} {/handle_tell {%1} {%2}}
```
That trigger is unlikely to work in practice however since most blocks end with a prompt.

Not a bad idea to trigger multi-line mode on %a. It's not really useful for anything else, that I know of. %a predates multi-line mode, so I'll probably start out with a warning not to use it in single-line triggers.

As for matching newlines as spaces, great idea. I've added it to my todo list, but I'm not sure what a good way to enable that mode for a trigger is. This would probably be a little bit better:
```
#act {line\scontaining\ssome\stext}
```
